src/plasma/plasma_manager.cc

```diff
@@ -832,6 +843,16 @@ void process_data_request(event_loop *loop,
                           int64_t data_size,
                           int64_t metadata_size,
                           ClientConnection *conn) {
+
+  /** Monitor objects that are in progress of being received.
+   *  If a read fails while receiving this object, its
+   *  ObjectID will be removed. If the object is successfully
+   *  received, its ObjectID is removed by process_add_object_notification.
+   *  If a shared buffer for the object cannot be created,
+   *  then the receive is ignored, and the corresponding ObjectID
+   *  is removed from receives_in_progress in this function. */
+  conn->manager_state->receives_in_progress.insert(object_id);
```

Comment:
Maybe move this line to the if-else block at the end of this function so that we don't have to erase?